Source/qol/stash.cpp

```diff
@@ -658,7 +654,11 @@ void GoldWithdrawNewText(string_view text)
 			int newGoldValue = WithdrawGoldValue * 10;
 			newGoldValue += digit;
 			if (newGoldValue <= InitialWithdrawGoldValue) {
+				// update to newly set value
 				WithdrawGoldValue = newGoldValue;
+			} else {
+				// lazily 'set & cap' to maximum gold value
+				WithdrawGoldValue = InitialWithdrawGoldValue;
 			}

```

Comment:
Can simplify this using `min`

```suggestion
			// Mimic D2 stash behaviour, cap withdrawal amounts to maximum gold value
			WithdrawGoldValue = std::min(newGoldValue, InitialWithdrawGoldValue);
```